### ISSUE TITLE
Striping headers breaks linux compatibility.

### ISSUE THREAD

**Issue by bartoszek** [cnr-isti-vclab/vcglib#17]
f2bbdb787adb20a1a941bc9a6cf4780e1f2ea1ec
This commit strips some headers witch in results in error when building meshlab on linux system.
Error message:
```
../../../vcglib/vcg/container/simple_temporary_data.h:52:64: error: ‘memcpy’ was not declared in this scope
         if(datasize!=0) memcpy(newdataLoc,data,sizeof(datasize));
                                                                ^
../../../vcglib/vcg/container/simple_temporary_data.h: In member function ‘void vcg::VectorNBW<bool>::resize(const int&)’
../../../vcglib/vcg/container/simple_temporary_data.h:64:57: error: ‘memset’ was not declared in this scope
         memset(&data[oldDatasize],0,datasize-oldDatasize);
```

**Comment by cignoni** [comment]
Compiling what file?

**Comment by bartoszek** [comment]
Error arise when compiling 'common.pro'
```
cd common/ && ( test -e Makefile || /usr/bin/qmake -o Makefile /home/bartus/AUR/meshlab-git/src/meshlab/src/common/common.pro ) && make -f Makefile 
make[1]: Wejście do katalogu '/home/bartus/AUR/meshlab-git/src/meshlab/src/common'
g++ -c -pipe -Wno-unknown-pragmas -Wno-unknown-pragmas -O2 -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong -std=gnu++11 -Wall -W -D_REENTRANT -fPIC -DMESHLAB_SCALAR=float -DGLEW_STATIC -DQT_NO_DEBUG -DQT_OPENGL_LIB -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_XML_LIB -DQT_XMLPATTERNS_LIB -DQT_NETWORK_LIB -DQT_SCRIPT_LIB -DQT_CORE_LIB -I. -I../.. -I../../../vcglib -I../../../vcglib/eigenlib -I../external/glew-1.7.0/include -I. -I../external/jhead-2.95 -isystem /usr/include/qt -isystem /usr/include/qt/QtOpenGL -isystem /usr/include/qt/QtWidgets -isystem /usr/include/qt/QtGui -isystem /usr/include/qt/QtXml -isystem /usr/include/qt/QtXmlPatterns -isystem /usr/include/qt/QtNetwork -isystem /usr/include/qt/QtScript -isystem /usr/include/qt/QtCore -I. -isystem /usr/include/libdrm -I/usr/lib/qt/mkspecs/linux-g++ -o moc_meshmodel.o moc_meshmodel.cpp
In file included from ../../../vcglib/vcg/complex/complex.h:42:0,
                 from ml_mesh_type.h:4,
                 from meshmodel.h:32,
                 from moc_meshmodel.cpp:9:
../../../vcglib/vcg/container/simple_temporary_data.h: In member function ‘void vcg::VectorNBW<bool>::reserve(const int&)’
../../../vcglib/vcg/container/simple_temporary_data.h:53:64: error: ‘memcpy’ was not declared in this scope
         if(datasize!=0) memcpy(newdataLoc,data,sizeof(datasize));
                                                                ^
../../../vcglib/vcg/container/simple_temporary_data.h: In member function ‘void vcg::VectorNBW<bool>::resize(const int&)’
../../../vcglib/vcg/container/simple_temporary_data.h:65:57: error: ‘memset’ was not declared in this scope
         memset(&data[oldDatasize],0,datasize-oldDatasize);
```

**Comment by bartoszek** [comment]
After rolling back changes to `simple_temporary_data.h` I've got more similar errors form `edit_align.pro`
http://pastebin.com/5KTLrsuq

**Comment by cignoni** [comment]
Ok, probably found. 
GCC is compiling a path that is never executed and that clang strip out...
`meshlabmini.pro` compiles correctly. (meshlabfull just started and still going...)

**Comment by bartoszek** [comment]
Yes, libcommon compiles correctly for me now. As for `edit_align.pro` same error.

**Comment by cignoni** [comment]
Can you try again with a clean freshly checked out repo? 
On my linux box it compiles...

**Comment by bartoszek** [comment]
Still the same error in `edit_align.pro`, compiled in `clean chroot_environment`.
// _my system:_ `arch-linux(4.10.2), gcc-6.3, qt-5.8`
// _system libraries used instead of externals ones:_ `muparser-2.2.5, lavmar-2.6, lib3ds-1.3, mpir-2.7.2`
// _edit_align.pro build log_ [{pastebin}](http://pastebin.com/J1TpT2cP)
// _full_build_log_ [{dropbox}](https://www.dropbox.com/s/sz8mcf06ulbh7d8/meshlab-git-2016.12.r96.gb176abe8-1-x86_64-build.log?dl=0)

**Comment by bartoszek** [comment]
@cignoni : Include `<vector>` in _histogram.h_ and meshlab build fine.

**Comment by bartoszek** [comment]
fixed in commit 718f5b182434afc0ce2defe18d8e1b30ac558bca
